PMMain/Midas/MidasGUI/MidasThread.cpp

```diff
@@ -41,3 +42,97 @@ void MidasThread::threadEmitStateString(std::string str)
 {
     emit emitStateString(str);
 }
+
+
+std::vector<ringData> MidasThread::readKeyboardSetupFile()
+{
+    
+    std::ifstream readRingData("C:/Users/Owen/Documents/keyboardData.txt");
+    std::string temp;
+    int vect_num = 0;
+    int ringInOutSwitch = 0; //0 = ringIn , 1 = ringOut 
+    bool holdkey_check = false;
+
+    ringData::keyboard_value key_temp; //lvl1
+    std::vector<ringData::keyboard_value> tempRingInData[100]; //lvl2
+    std::vector<ringData::keyboard_value> tempRingOutData[100]; //lvl2
+    ringData ringTemp; //lvl3
+    std::vector<ringData> keyboardData; //lvl4
+
+
+    //readRingData.open();
+    if (readRingData.is_open())
+    {
+        while (!readRingData.eof())
+        {
+            getline(readRingData, temp);
+            char* temp_char = (char*)malloc(sizeof(char)*(temp.length() + 1));
+            strcpy(temp_char, temp.c_str());
+
+            for (int i = 0; i < temp.length(); i++)
+            {
+
+                if (temp_char[i] != char(32) && holdkey_check == false)
```

Comment:
Hmm... If you want to check for space use ' ' instead. And you're only allowing one space between?